src/UserService.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace RegistrationForm;
+
+class UserService {
+    private $name;
+    private $last_name;
+    private $email;
+    private $phone;
+    private $password;
+    private $confirmed_password;
+
+    public function __construct(string $name, string $last_name, string $email, string $phone, string $password, string $confirmed_password){
```

Comment:
I propose you to:
-  move all these arguments to register() method
- remove class properties, which are related to user object
- create new User in register method not from UserService class properties, but from arguments which you'll provide in that method.
By doing that you'll make UserService more universal.
Imagine, what will happen if you'll need to have functionality in this class to change password for already existing user.
To do that you'll need to add another method like changePassword
In this case you'll not need all this data like name, email, phone. You'll need only password and confirmed_password. But in current implementation in order to create this class object you need to have this data. Where you'll get it?

When deciding what things to put in constructor, try to thing separate them in two categories. 1 - dependencies which are required for class to function. 2 - data which you're going to process.
First category should be included in constructor, second category should be passed to concrete methods.
Examples of things from first category - validators, another services, repositories. Basically classes which are doing something.
Things from second category usually are the ones against which these actions are performed - some data.